docs/content/docs/data-structures/task-manager.md

```diff
@@ -1,5 +1,135 @@
+
+### Consensus Based DDS
+An important note about TaskManager is that it is a consensus based DDS. This essentially means that ops are not accepted until every client acknowledge and accepts the op. This differs from an "optimistic" DDS (i.e. [SharedMap]({{< relref "./map.md" >}})) which immediately accept ops and then relays them to other clients. For more information regarding different types of DDSes, click [here]({{< relref "./overview.md" >}}).
+
+## Usage
+
+### APIs
+
+The `TaskManager` object provides a number of methods to manage the execution of tasks. Please note: each API requires an input of `taskId` which is type `string`.
+
+
+- `volunteerForTask(taskId)` -- Adds the client to the task queue **once**. It returns a promise that resolves `true` if the client is assigned the task and `false` if the task was completed by another client. It will throw an error if the client disconnects while in queue.
+- `subscribeToTask(taskId)` -- Will continuously add the client to the task queue. Does not return a value, and will therefore require lisnening to [events](#events) to determine if the task is assigned, lost, or completed.
+- `subscribed(taskId)` -- Returns a boolean to indicate if the client is subscribed to the task.
+- `complete(taskId)` -- Will release all clients from the task queue, including the currently assigned client.
+- `abandon(taskId)` -- Exits the queue and releasing the task if currently assigned. Will also unsubscribe from the task (if subscribed).
+- `queued(taskId)` -- Returns a boolean to indicate if the client is in the task queue (being assigned a task is still considered queued).
+- `assigned(taskId)` -- Returns a boolean to indicate if the client is assigned the task.
+
+### Events
+
+`TaskManager` is an `EventEmitter`, and will emit events when a task is assigned to the client or released. Each of the below events fire with an event listener that contains a callback argument `taskId`. This represents the task for which the event was fired for.
+
+- `assigned` -- Fires when the client reaches the top of the task queue and is assigned the task.
+- `lost` -- Fires when the client disconnects while being assigned a task and is removed from the queue.
+- `completed` -- Fires when the assigned client successfully calls `complete()`.
+
+### Creation
+
+To create a `TaskManager`, call the static create method:
```

Comment:
Either link to a reference topic for the `create` method or explain what the two parameters are.